scripts/cli/CodeCompass.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+import getpass
+import os
+import sys
+import resource
+
+import handler.parse
+import handler.server
+import handler.workspace
+from util.util import *
+
+#--- Argument parser ---#
+
+parser = argparse.ArgumentParser()
+subparsers = parser.add_subparsers()
+
+#--- Parse handling options ---#
+
+parse_parser = subparsers.add_parser('parse', help = 'Parser handling')
+
+parse_parser.add_argument(
+  '-w', '--workdir', type = str, required = True,
+  help = 'Path of workspace directory of the projects in which config files are\
+  stored.')
+parse_parser.add_argument(
+  '-n', '--name', type = str, required = True,
+  help = 'Project name to identify the project.')
+parse_options = parse_parser.add_mutually_exclusive_group()
+parse_options.add_argument(
+  '-b', '--build', type = str,
+  help = 'This option is used to build the project and create build logs. This\
+  flag has to be given the build command which builds the project.')
+parse_options.add_argument(
+  '-l', '--logfile', type = str,
+  help = 'Path of the parse log which is used by the parser as input.')
+parse_options.add_argument(
+  '-o', '--logonly', type = str,
+  help = 'Log only. In this case parser is not started after logging session.')
+parse_parser.add_argument(
+  '-f', '--force', action = 'store_true',
+  help = 'If -l is given and the log file already exists or -b is given and the\
+  project directory already exists then the commands are not executed unless\
+  this flag is given. In case of forcing the log file will be overwritten and\
+  project directory will be reused.')
+parse_parser.add_argument(
+  '-c', '--crash', action = 'store_true',
+  help = 'If this flag is given then parser stops when the parser crashes.')
+parse_parser.add_argument(
+  '-i', '--input', nargs='*', 
+  help = 'The input of the parsers can be a compilation database (see:\
+  http://clang.llvm.org/docs/JSONCompilationDatabase.html) or a path of a\
+  directory under which the files will be consumed. Here you can list the\
+  compilation databases and the paths: -i /path/one -i /path/two -i\
+  compilation_database.json')
+parse_parser.add_argument(
+  '--loglevel', choices = ['debug', 'info', 'warning', 'error', 'critical'],
+  help = 'Logging level of the parser.')
+parse_parser.add_argument(
+  '--labels', type = str, default = '',
+  help = 'Name paths with labels. Some CodeCompass parser components will parse\
+  only files under these paths. Furthermore on the GUI there will be shortcuts\
+  on these paths. The string format has to be:\
+  label1=/path/one:label2=/path/two. If not given then an interactive interface\
+  will ask for the paths.')
+parse_parser.add_argument(
+  '--noindex', action = 'store_true',
```

Comment:
In this script we remove indices before parse and add the indices after the parse. Indices allow the database to find the data fast. If we allow this flag, database operations will be slow. So I don't think we should support this flag.